### PR TITLE
Branch cc 2

### PR DESCRIPTION
**Pull request by liewchooichin**
I added in simple test cases. The files changed are listed below.

Using **branch-cc-2**.

$ git status
On branch main
Your branch is up to date with 'origin/main'. The api key is still being tracked. 

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   src/main/java/sg/edu/ntu/bus_api/service/ApiKey.java
        modified:   src/main/resources/application.properties

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        src/test/java/sg/edu/ntu/bus_api/AppBusArrivalControllerTest.java
        src/test/java/sg/edu/ntu/bus_api/AppBusRouteControllerTest.java
        src/test/java/sg/edu/ntu/bus_api/AppBusServiceControllerTest.java
        src/test/java/sg/edu/ntu/bus_api/AppBusStopControllerTest.java
        src/test/java/sg/edu/ntu/bus_api/TestNotes.md

no changes added to commit (use "git add" and/or "git commit -a")

### CODE REVIEW

**Comment by liewchooichin** [comment]
Change the **application.properties** to persist the database. Use hibernate.dll.auto=update. IMPORTANT: Must update the application.properties to make the database persist.

I remove the DataLoader by renaming its extention to .xxx. The data is too large to be loaded at once. Instead use the GET /datamall to load the table one by one.

The front-end point findAll is too large. So I replaced it with findFirstOne.  This can be used as a test to make sure the endpoint is working.

Add in the following GET:
- GET bus service by serviceNo.
- GET bus stops by busStopCode.
- Get bus route by serviceNo and direction.

File: bus-api/src/main/java/sg/edu/ntu/bus_api/config/DataLoader.java.xxx
Comment (agentjimlam) [inline bus-api/src/main/java/sg/edu/ntu/bus_api/config/DataLoader.java.xxx:1]: okay, so this file becomes a text file. i see

**Review by agentjimlam — APPROVED** [review]
Noted the changes made to Controller, Service, and BusRouteRepository is for doing the test files 

Noted you've added custom JPA query in BusRouteRepository.java to find Bus Route by service and ignore direction of bus part

The front-end point findAll is too large for testing purposes. So in controller java files, the web handlers for findAll replaced with findFirstOne, where it now returns the first id object instead of the whole Bus Stop object for example.

**Review by agentjimlam — COMMENTED** [review]
Also noted that application.properties changes to the setting for jpa-hibernate.ddl-auto from =create to =update is important to make database persist

**Pull request merged**